submit_cycle.sh

```diff
@@ -16,57 +16,73 @@
 # -decide how to manage soil moisture DA. Separate DA script to snow? 
 # -add ensemble options
 
+File_setting=$1
 ############################
-# experiment name 
-
-exp_name=openloop_testing
-
-############################
-# model options
-
-export ensemble_size=1 # ensemble_size of 1 = do not run ensemble 
-                       # LETKF-OI pseudo ensemble uses 1
-
-atmos_forc='gdas' # options: gdas, gswp3, gefs_ens
-
-dates_per_job=2 # number of cycles to submit in a single job
-
-############################
-# DA options
-
-# select YES or NO
-export do_DA=YES  # do full DA update
-do_hofx=NO  # use JEDI to calculate hofx, but do not do update 
-             # only used if do_DA=NO  
-export ASSIM_IMS=NO
-export ASSIM_GHCN=YES
-export ASSIM_SYNTH=NO
-
-DAtype="letkfoi_snow" # options: "letkfoi_snow" , "letkf_snow"
+# read in DA settings for the experiment
+while read line
+do
+    [[ -z "$line" ]] && continue
+    [[ $line =~ ^#.* ]] && continue
+    key=$(echo ${line} | cut -d'=' -f 1)
+    value=$(echo ${line} | cut -d'=' -f 2)
+    case ${key} in
+        "CYCLEDIR")
+        CYCLEDIR=${value}
+        ;;
+        "EXPDIR")
+        EXPDIR=${value}
+        ;;
+        "exp_name")
+        exp_name=${value}
+        ;;
+        "ensemble_size")
+        ensemble_size=${value}
+        ;;
+        "atmos_forc")
+        atmos_forc=${value}
+        ;;
+        "dates_per_job")
+        dates_per_job=${value}
+        ;;
+        "do_DA")
+        do_DA=${value}
+        ;;
+        "WORKDIR")
+        WORKDIR=${value}
+        ;;
+        "DIFF_CYCLEDIR")
+        DIFF_CYCLEDIR=${value}
+        ;;
+        "ICSDIR")
+        ICSDIR=${value}
+        ;;
+        #default case
+        #*)
+        #echo ${line}
+        #;;
+    esac
+done < "$File_setting"
 
 ############################
 # set your directories
-
-CYCLEDIR=$(pwd)  # this directory
-export WORKDIR=/scratch2/BMC/gsienkf/Clara.Draper/workdir/ # temporary work dir 
-export OUTDIR=${CYCLEDIR}/exp_out/${exp_name}/output/      # directory where output will be saved
-ICSDIR="/scratch2/BMC/gsienkf/Clara.Draper/DA_test_cases/offline_ICS/single/" # OUTDIR for experiment with initial conditions
-                                                           # will use ensemble of restarts if present, otherwise will try 
-                                                           # to copy a non-ensemble restart into each ensemble restart
+if [[ -z "$CYCLEDIR" ]]; then
```

Comment:
The idea here should be that CYCLEDIR defaults to the current directory. I'm not sure why I suggested using DIFF_CYCLEDIR, but looking at this code, I don't think we need it. We can just use this logic : if cycledir is provided in the input, use it, if not use current directory. Right? 

Note: we need to make sure that this works OK when the script re-submits itself.